backend/src/main/java/com/google/sps/notifs/EmailNotifier.java

```diff
@@ -0,0 +1,178 @@
+        return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+    }
+
+    /**
+     * Create a MimeMessage using the parameters provided.
+     *
+     * @param to email address of the receiver
+     * @param subject subject of the email
+     * @param bodyText body text of the email
+     * @return the MimeMessage to be used to send email
+     * @throws MessagingException
+     */
+    public static MimeMessage createEmail(String to, String subject, String bodyText)
+            throws MessagingException {
+
+        Properties props = new Properties();
+        Session session = Session.getDefaultInstance(props, null);
+
+        MimeMessage email = new MimeMessage(session);
+
+        email.setFrom(new InternetAddress(to));
+        email.addRecipient(javax.mail.Message.RecipientType.TO, new InternetAddress(to));
+        email.setSubject(subject);
+        email.setText(bodyText);
+        return email;
+    }
+
+    /**
+     * Create a message from an email.
+     *
+     * @param emailContent Email to be set to raw of message
+     * @return a message containing a base64url encoded email
+     * @throws IOException
+     * @throws MessagingException
+     */
+    public static Message createMessageWithEmail(MimeMessage emailContent)
+            throws MessagingException, IOException {
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+        emailContent.writeTo(buffer);
+        byte[] bytes = buffer.toByteArray();
+        String encodedEmail = Base64.encodeBase64URLSafeString(bytes);
+        Message message = new Message();
+        message.setRaw(encodedEmail);
+        return message;
+    }
+
+    public static void main(String[] args)
+            throws GeneralSecurityException, MessagingException, IOException {
+        EmailNotifier emailNotifier = new EmailNotifier("jordan", "grantjustice@google.com");
+        emailNotifier.notifyUser();
+    }
+
+    /**
+     * Getter function that returns the string representing the email recipients name .
+     *
+     * @return String representing the name of the email recipient.
+     */
+    @Override
+    public String getName() {
+        return super.getName();
+    }
+
+    /** Function that access its api and using it sends an email */
+    //    TODO(): Create a dummy email for ad lib itself to send emails.
+    public void notifyUser() throws MessagingException, IOException, GeneralSecurityException {
+
+        MimeMessage email =
+                createEmail(
+                        getToEmail(),
+                        "Ad-Lib Meeting Found",
+                        " Hey "
+                                + this.getName()
+                                + " Please Join your Ad-Lib meeting via the link below : \n"
+                                + " http://meet.google.com/new");
+        final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
```

Comment:
tiny tiny nit: There's no need to mark this as final. The compiler can tell which variables are "effectively final"

Reply:
Ack